chain/db/db.go

```diff
@@ -33,21 +35,42 @@ func NewDB(dialect, path string, metadata map[string]string) (*BandDB, error) {
 
 	db.AutoMigrate(
 		&Metadata{},
-		&Event{},
+		&Block{},
+		&Transaction{},
+		&Account{},
 		&Validator{},
 		&ValidatorVote{},
 		&DataSource{},
 		&DataSourceRevision{},
 		&OracleScript{},
 		&OracleScriptRevision{},
-		&Block{},
-		&Transaction{},
-		&Report{},
-		&ReportDetail{},
+		&RelatedDataSources{},
 		&Request{},
 		&RequestedValidator{},
 		&RawDataRequests{},
-		&RelatedDataSources{},
+		&Report{},
+		&ReportDetail{},
+	)
+
+	db.Model(&Block{}).AddForeignKey(
+		"proposer",
+		"validators(consensus_address)",
+		"RESTRICT",
+		"RESTRICT",
+	)
+
+	db.Model(&Transaction{}).AddForeignKey(
+		"block_height",
+		"blocks(height)",
+		"RESTRICT",
+		"RESTRICT",
+	)
+
+	db.Model(&Transaction{}).AddForeignKey(
```

Comment:
Is it possible that we add transaction before account and violate fk constraint?

Reply:
A sender is needed to declare before in chain (Cannot send a transaction without account number and sequence number)